### PR TITLE
[Spending controls] Create invite with initial allowance amount

### PR DESCRIPTION
**Pull request by malted**
image

### CODE REVIEW

**Comment by garyhtou** [comment]
@sampoder can you review this one?

File: app/models/organizer_position_invite.rb [hunk 1]
```diff
@@ -111,7 +112,17 @@ def accept(show_onboarding: true)
 
     self.accepted_at = Time.current
 
-    self.save
+    ActiveRecord::Base.transaction do
+      self.save
+
+      unless initial_control_allowance_amount_cents.nil?
```
Comment (sampoder) [inline app/models/organizer_position_invite.rb]:
```suggestion
      if initial_control_allowance_amount_cents.present?
```

File: app/models/organizer_position_invite.rb [hunk 2]
```diff
@@ -111,7 +112,17 @@ def accept(show_onboarding: true)
...
+        # Create control
+        organizer_position.spending_controls.create!
+
+        # Create allowance
+        organizer_position.active_spending_control.allowances.create!(authorized_by_id: sender_id, amount_cents: initial_control_allowance_amount_cents, memo: "Initial control")
```
Comment (sampoder) [inline app/models/organizer_position_invite.rb]:
```suggestion
        organizer_position.active_spending_control.allowances.create!(authorized_by_id: sender_id, amount_cents: initial_control_allowance_amount_cents, memo: "Initial allowance")
```

File: app/controllers/organizer_position_invites_controller.rb [hunk 3]
```diff
@@ -21,7 +21,10 @@ def create
     role = invite_params[:role]
     is_signee = invite_params[:is_signee] || false
 
-    service = OrganizerPositionInviteService::Create.new(event: @event, sender: current_user, user_email:, is_signee:, role:)
+    enable_spending_controls = (invite_params[:enable_controls] == "true") && (role != "manager")
```
Comment (sampoder) [inline app/controllers/organizer_position_invites_controller.rb]: Would it make sense to error if someone attempts to enable spending controls for a manager?

File: app/services/organizer_position_invite_service/create.rb [hunk 4]
```diff
@@ -2,20 +2,23 @@
...
+      @enable_spending_controls = enable_spending_controls
+      @initial_control_allowance_amount = initial_control_allowance_amount
 
       args = {}
       args[:event] = @event
       args[:sender] = @sender
       args[:initial] = @initial
       args[:is_signee] = @is_signee
       args[:role] = @role if role
+      args[:initial_control_allowance_amount_cents] = @enable_spending_controls ? (@initial_control_allowance_amount.to_f * 100).to_i : nil
```
Comment (sampoder) [inline app/services/organizer_position_invite_service/create.rb]: can we use `Monetize.parse`?

**Comment by sampoder** [comment]
Would it make sense to include this in the mailer on invite?

File: app/controllers/organizer_position_invites_controller.rb (hunk 3 above)
Comment (malted) [reply app/controllers/organizer_position_invites_controller.rb]: Added validation on opi

**Pull request merged**